analyse.js

```diff
@@ -0,0 +1,58 @@
+const readline = require('readline'),    
+    rl = readline.createInterface({
+        input: process.stdin, 
+        output: process.stdout 
+    }),
+    fs = require('fs');
+
+const EagleAnalyse = (function () {
+    function Statistic(logFile) {   
+        let count = 0, win = 0, winC = 0, loos = 0, loosC = 0;
+        if (logFile) {
+            fs.exists(logFile, function (exists) {
+                if (exists) {
+                    fs.createReadStream("logData.txt")
+                        .on('data', function (chunk) {
+
+                            let arr = chunk.toString().split('\r\n'),
+                                arrWins = chunk.toString().replace(/(?:\\[rn]|[\r\n]+)+/g, "").split('0'),
+                                arrLoos = chunk.toString().replace(/(?:\\[rn]|[\r\n]+)+/g, "").split('1');
+
+                            arrLoos.map(function(i){
+                                loosC = (i.length > loosC) ? i.length : loosC;
+                            });
+                            arrWins.map(function(i){
+                                winC = (i.length > winC) ? i.length : winC;
+                            });
+
+                            count = arr.length - 1;
+                            arr.forEach(function(i){
```

Comment:
В таких выражениях удобнее использовать функцию стрелку. Кроме прочего, попробуйте вместо forEach использовать reduce
```suggestion
                            arr.forEach((i) => {
```